### PR TITLE
Implemented firebase authentication, First steps for issue #98

### PR DESCRIPTION
**Pull request by ravi-kishan**
First steps for issue #98

Changes: Firebase authentication implemented
TODO ->Injector factory for created activity

Screenshots for the change:

### CODE REVIEW

File: app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java [hunk 1]
```diff
@@ -14,7 +17,13 @@ public AuthViewModel() {
 
     }
 
+    private static final FirebaseAuthLiveData firebaseAuthLiveData = new
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java]: Move this variable declaration before the constructor.

File: app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java [hunk 2]
```diff
@@ -14,7 +17,13 @@ public AuthViewModel() {
...
+            FirebaseAuthLiveData();
+
+    public static LiveData<FirebaseUser> getFirebaseAuthLiveData() {
+        return firebaseAuthLiveData; }
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java]: Move the `}` to the new line. Follow the conventions used in the project.

File: app/src/main/java/com/iitp/njack/iitp_connect/core/home/FirebaseAuthLiveData.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import com.google.firebase.auth.FirebaseUser;
+
+public class FirebaseAuthLiveData extends LiveData<FirebaseUser> {
+
+    private FirebaseAuth firebaseAuth = FirebaseAuth.getInstance();
+    private FirebaseAuth.AuthStateListener authStateListener = new FirebaseAuth.AuthStateListener() {
+                @Override
+                public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
+                    FirebaseUser firebaseUser = firebaseAuth.getCurrentUser();
+                    setValue(firebaseUser);   }
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/home/FirebaseAuthLiveData.java]: Move `}` to the new line.

File: app/src/main/java/com/iitp/njack/iitp_connect/core/home/HomeActivity.java [hunk 4]
```diff
@@ -52,6 +64,24 @@ protected void onCreate(Bundle savedInstanceState) {
         binding.navView.setNavigationItemSelectedListener(this);
 
         drawerNavigator = new DrawerNavigator(this, authViewModel);
+
+
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/home/HomeActivity.java]: too many blank lines.

File: app/src/main/java/com/iitp/njack/iitp_connect/core/home/HomeActivity.java [hunk 5]
```diff
@@ -52,6 +64,24 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            @Override
+            public void onChanged(@Nullable FirebaseUser firebaseUser) {
+                if (firebaseUser == null) {
+                    final Intent intent = AuthUI.getInstance().createSignInIntentBuilder()
+                            .build();
+                    startActivityForResult(intent, RC_SIGN_IN);
+                } else {
+                   // TODO Create UI for logged in user
+                }
+
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/home/HomeActivity.java]: Remove the blank line.

**Review by dr0pdb — CHANGES_REQUESTED** [review]
Minor formatting changes required. Rest looks good.

**Review by dr0pdb — APPROVED** [review]

**Comment by dr0pdb** [comment]
@ravi-kishan Should I merge this or do you want to add more commits to this PR itself?

**Comment by ravi-kishan** [comment]
I want to add the injector factory before merging.

**Comment by ravi-kishan** [comment]
The branch can be merged now

File: app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/android/ActivityBuildersModule.java [hunk 6]
```diff
@@ -17,4 +20,13 @@
...
     abstract HomeActivity contributeHomeActivity();
+
+    @ContributesAndroidInjector
+    abstract KickoffActivity contributeKickoffActivity();
+
+    @ContributesAndroidInjector
+    abstract EmailActivity contributeEmailActivity();
+
+    @ContributesAndroidInjector
+    abstract CredentialSaveActivity contributeCredentialSaveActivity();
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/android/ActivityBuildersModule.java:31]: I don't see these activities defined in the code? Are they automatically created by the Firebase SDK?

Comment (ravi-kishan) [reply app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/android/ActivityBuildersModule.java:31]: The activities are part of firebaseUI library.

Comment (ravi-kishan) [reply app/src/main/java/com/iitp/njack/iitp_connect/common/di/modules/android/ActivityBuildersModule.java]: They are not automatically created,but app could not run without injecting them.

**Review by sheetal-g9 — COMMENTED** [review]
Reviewing  #118 
Sign in and Sign out working fine without any errors.Checked app functioning with debugger .
working well.

File: app/src/main/java/com/iitp/njack/iitp_connect/core/home/HomeActivity.java (hunk 4 above)
Comment (sheetal-g9) [reply app/src/main/java/com/iitp/njack/iitp_connect/core/home/HomeActivity.java]: 'java.util.collections' is never used,can be removed.

**Review by dr0pdb — APPROVED** [review]

**Pull request merged**